.github/workflows/ci.yml

```diff
@@ -0,0 +1,78 @@
+name: Build & Tests
+
+on:
+  push:
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_test:
+    # Used for installing Miri
+    # https://rust-lang.github.io/rustup-components-history/
+    env:
+      TOOLCHAIN_VERSION: '2022-09-26'
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        channel: [ "1.56.1", "stable", "beta", "nightly" ]
+        target: [ "i686-unknown-linux-gnu", "x86_64-unknown-linux-gnu", "arm-unknown-linux-gnueabi", "aarch64-unknown-linux-gnu", "powerpc-unknown-linux-gnu", "powerpc64-unknown-linux-gnu", "wasm32-wasi" ]
+        features: [ "" , "--features alloc,simd,simd-nightly"]
+
+    # Prevents a workflow run from failing when a job fails
+    # If target is x86 or x86_64, 
+    # or we are using simd-nightly while not on the nightly channel
+    # do not continue on error (so crash on error)
+    continue-on-error: ${{ !(contains(matrix.target, 'x86_64') || contains(matrix.target, 'i686')) }}
```

Comment:
Unfortunately I just realized that this isn't sufficient. If `cargo check`, `cargo build`, or `cargo miri test` fail, we should report those failures regardless of what target we're building for. It's only `cargo test` that should be allowed to fail on non-x86_64/non-i686. Is there a way to test for which command we're running?

Reply:
Not that I'm aware of, you can still inspect the tasks by browsing through the workflow runs though

Reply:
Oh wait it looks like we don't need this at all! You're already only running the `Run ${{ matrix.channel }} tests` job conditionally:

```text
if: ${{ contains(matrix.target, 'x86_64') || contains(matrix.target, 'i686') }}
```

Reply:
So we should be able to just remove this entirely. I can add it back later if we add more nightly features or something.

Reply:
Ah alright!

Reply:
Just to be sure, you mean remove the `if` check you just pasted right?

Reply:
No sorry, I mean remove this `continue-on-error` line (including the comment on it and the commented-out `continue-on-error` line below it). IIUC, the `if` in `Run ${{ matrix.channel }} tests` should be all that we need.